libraries/chain/hotstuff/qc_chain.cpp

```diff
@@ -56,7 +56,7 @@ namespace eosio::chain {
       const auto& finalizers = _pacemaker->get_finalizer_policy().finalizers;
 
       for (size_t i = 0; i < finalizers.size();i++) {
-         if (finalizers[i].public_key == finalizer_key) {
+         if ((finalizers[i].public_key <=> finalizer_key) == 0) {
```

Comment:
Why not just `==` ?

Reply:
Because `fp` defines a customized `<=>` https://github.com/AntelopeIO/bls12-381/blob/b179f9b2724cc7faafb7d9ec920c2009577e842e/include/bls12-381/fp.hpp#L82, that makes compiling failing:

```
.../libraries/chain/hotstuff/qc_chain.cpp:361:50: error: use of deleted function ‘constexpr bool fc::crypto::blslib::bls_public_key::operator==(const fc::crypto::blslib::bls_public_key&) const’
  361 |             if (finalizers[i].public_key == vote.finalizer_key) {
...
.../libraries/libfc/include/fc/crypto/bls_public_key.hpp:30:15: error: use of deleted function ‘constexpr bool bls12_381::g1::operator==(const bls12_381::g1&) const’
...
.../libraries/libfc/libraries/bls12-381/include/bls12-381/g.hpp:23:8: error: no match for ‘operator==’ (operand types are ‘bls12_381::fp’ and ‘bls12_381::fp’)
   23 |     fp x;
```

Reply:
https://stackoverflow.com/questions/58780829/non-defaulted-operator-doesnt-generate-and

I'll get a fix into bls12-381 for this.

Reply:
https://github.com/AntelopeIO/bls12-381/pull/21

Reply:
You can update to the latest `hotstuff_integration` and `==` should work.